Basic-data-structure: add Makefile, source file

vladexe opened the pull request:
Adds basic data structure kernel list, use kobject with sysfs

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

ekovalyov committed:
Create README.md

marynasergiyenko committed:
Merge pull request #1 from ekovalyov/patch-1

bash homework

marynasergiyenko committed:
Merge pull request #2 from ekovalyov/patch-2

git homework

ekovalyov committed:
module homework

alexposukhov committed:
Merge pull request #29 from ekovalyov/homework_module

module homework

ekovalyov committed:
homework: basic data structures

vladexe committed:
Basic-data-structure: add Makefile, source file

Adds basic data structure kernel list, use kobject with sysfs

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

vladexe committed:
Basic-data-structure: change the main.c

Changes main.c.Adds some error managment. Such as error with
creating kobject or sysfs file

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

vladexe committed:
Basic-data-structure: adds checkpatch.log and test scripts

Adds checkpatch log which was generated.And some scripts for
test module

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

Anton-Soroka commented on 04_basic_struct/main.c:
```diff
@@ -0,0 +1,100 @@
+
+	list_for_each_entry(item, &list, head) {
+		pr_info("Try to read some info\n");
+		state = sprintf(buf + offset, "%s", item->str);
```
better to use snprintf and specify amount of free space left in buffer: (PAGE_SIZE - offset)

Anton-Soroka commented on 04_basic_struct/main.c:
```diff
@@ -0,0 +1,100 @@
+
+static int mymodule_init(void)
+{
+	int state = 0;
+
+	pr_info("Module initialized successfully\n");
+	hello = kobject_create_and_add("kobject_list", kernel_kobj);
+	if (hello == NULL) {
+		pr_info("impossible to create Kobject!\n");
+		state = -EINVAL;
+	} else {
+		state = sysfs_create_file(hello, &attr.attr);
+		if (state == -EINVAL) {
+			kobject_put(hello);
+			pr_info("impossible to create File!\n");
+		}
+	}
+	return state;
```
better to use return on errors when possible
this way you can simplify the code
remove "else" and brackets and just put following code on the same indentation level

one more thing
pr_info("Module initialized successfully\n");
should be at the end, just before return 0

vladexe committed:
Basic-data-structure: adds kernel log after emulating

adds kernel log file, now uses snprintf

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

vladexe committed:
Basic-data-structure: adds kernel log after emulating

adds kernel log file, now uses snprintf

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

vladexe committed:
Basic-data-structure: adds command in Makefile

Adds "check" command in Makefile and add "return -EINVAL"
after error messages

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

vladexe committed:
Basic-data-structure: fix issue with -EINVAL

Signed-off-by: Vladyslav Andrishko <v.andrishko.v333@gmail.com>

Anton-Soroka reviewed (approved):
Looks good!
Please merge!
